src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
Could we do these reorderings inside `Docs` instead? The ExternsFile format should still contain all the DeclarationRefs.

- This way we don't slow down the existing compilation process
- The implicit requirement that these declarations have to be sorted are similar to the topo sort ordering requirement on ExternsFiles, they are not document/enforced (in the types) and cause weird errors.